rust_src/src/buffers.rs

```diff
@@ -209,6 +210,18 @@ impl LispBufferRef {
             c_int::from(self.fetch_byte(n))
         }
     }
+
+    #[inline]
+    pub fn overlays_before(&self) -> Option<LispOverlayRef> {
+        let ov = unsafe { bget_overlays_before(self.as_ptr()) };
+        unsafe { ov.as_ref().map(|o| mem::transmute(o)) }
+    }
```

Comment:
Perhaps these two function bodies could be refactored into a `new_from_ptr` method on `LispOverlayRef`?

    pub fn overlays_before(&self) -> Option<LispOverlayRef> {
        LispOverlayRef::new_from_ptr(unsafe { bget_overlays_before(self.as_ptr()) })
    }

Maybe?

Reply:
Would `from_ptr` also be ok ?

Reply:
Sounds reasonable.